src/activities/activity.types.ts

```diff
@@ -44,46 +58,116 @@ export class ActivityTypesHelper {
     }
   }
+
+  /**
+   * Get's back the activity group an activity belongs to or returns unspecified activity group
+   * @param activityType
+   * This function can also be called: Fighting with a non functional language
+   */
+  static getActivityGroupForActivityType(activityType: ActivityTypes): ActivityTypeGroups {
+    return ActivityTypeGroups[<keyof typeof ActivityTypeGroups>ActivityTypesHelper.getActivityTypeGroupsAsUniqueArray().find(activityTypeGroupString => { // Could also iterate over the map
+      return ActivityTypesGroupMapping.map[ActivityTypeGroups[<keyof typeof ActivityTypeGroups>activityTypeGroupString]].find((groupItem: ActivityTypes) => groupItem === activityType)
+    }) || ActivityTypeGroups.Unspecified];
+  }
+
 }
 
 /**
  * This enum works like a all matchers for normalized sport types between different naming across services
+ *
+ * It helps as you can call request an activity type with different namin eg .BackCountrySki or .BackCountrySkiing and get a uniform value
+ * Also helps in case you have persited data that do not match or have been peristed wrongly
+ *
+ * Important: don't forget to declare the original string value aka: 'Running' = 'Running'
+ *
+ * @todo how do we write activity names? Is it Alpine skiing? or Alpine Skiing? For now I used uppercase each word
  */
 export enum ActivityTypes {
-  'unknown' = 'Unknown sport',
-  'Unknown sport' = 'Unknown sport',
-  'undefined' = 'Unknown sport',
+  /**
+   * Unknown sport
+   */
+  'unknown' = 'Unknown Sport',
+  'Unknown sport' = 'Unknown Sport',
+  'Unknown Sport' = 'Unknown Sport',
+  'undefined' = 'Unknown Sport',
+  'Not specified sport' = 'Unknown Sport',
+  /**
+   * Other
+   */
+  'Other' = 'Other',
+  /**
+   * Generic
+   */
   'generic' = 'Generic',
   'Generic' = 'Generic',
+  /**
+   * Transition
+   */
   'transition' = 'Transition',
   'Transition' = 'Transition',
+  /**
+   * Fitness Equipment
+   */
   'fitness_equipment' = 'Fitness Equipment',
   'Fitness Equipment' = 'Fitness Equipment',
-  'Not specified sport' = 'Unknown sport',
-  'Other' = 'Unknown sport',
+
+  /**
+   * Multisport
+   */
   'Multisport' = 'Multisport',
+  'MultiSport' = 'Multisport',
   'multisport' = 'Multisport',
+
+  /**
+   * Virtual Running
+   */
   'running_virtual_activity' = 'Virtual Running',
   'VirtualRun' = 'Virtual Running',
+  'Virtual running' = 'Virtual Running',
+  'Virtual Running' = 'Virtual Running',
+  'VirtualRunning' = 'Virtual Running',
+  /**
+   * Running
+   */
   'Run' = 'Running',
   'run' = 'Running',
   'running_track' = 'Running',
   'running_trail' = 'Trail Running',
   'Running' = 'Running',
   'running' = 'Running',
+  'running_street' = 'Running',
+  'running_road' = 'Running',
+  /**
+   * Trail Running
+   */
+  'TrailRunning' = 'Trail Running',
+  'Trail Running' = 'Trail Running',
+  'Trail running' = 'Trail Running',
+  'trail_running' = 'Trail Running',
+  'trail' = 'Trail Running', // @todo hack?
```

Comment:
What's your source list of sports? I mean where did you see this only `trail` value?

Reply:
The specific I dont remember and better to go away I think. I ll remove this. 

The list is from the "Fit" file, other types of files (Suunto json, gpx, tcx etc). 

I had sentry alert me if "not found" so I was adding all those missing. 

Many of those are though "hand made" eg `TrailRunning` just to be able to acess the enum as `at.TrailRunning` and not with the brackets. :-) I am not sure 100% if the approach I am taking is the best in class here. It looks a bit dirty and "catch em all".... Just saying....